### PR TITLE
AC_Fence: add prearm check if fences enabled but none selected

### PR DESCRIPTION
**Pull request by Hwurzburg**
if this violates proper form, then i can move the entire init() to the fence.cpp file

### CODE REVIEW

**Comment by Hwurzburg** [comment]
@peterbarker not sure why it fails to build the MatekF405-Wing...it has plenty (relatively speaking) of room.. this take ~75bytes and there is almost 400b bytes left in the F405-Wing....admittedly, we need that build server sooner than later!

**Comment by rmackay9** [comment]
@Hwurzburg, this is where the new check should go: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AC_Fence/AC_Fence.cpp#L229

I think it should be at the top of this method.

**Comment by joshanne** [comment]
I agree with Randy here, this should really be a pre-arm check first.

File: libraries/AC_Fence/AC_Fence.h [hunk 1]
```diff
@@ -57,6 +57,9 @@ class AC_Fence
 
     void init() {
         _poly_loader.init();
+        if ( enabled() && !present() ) {  //warn if fences are enabled but none selected
+            GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "Fences enabled,but none selected!");
```
Comment (joshanne) [inline libraries/AC_Fence/AC_Fence.h:61]:
```suggestion
            GCS_SEND_TEXT(MAV_SEVERITY_WARNING, "Fences enabled, but none selected!");
```

**Comment by Hwurzburg** [comment]
changed to pre-arm as requested in DEVCALL....SITL tested...

File: libraries/AC_Fence/AC_Fence.cpp [hunk 2]
```diff
@@ -282,7 +282,13 @@ bool AC_Fence::pre_arm_check(const char* &fail_msg) const
...
     fail_msg = nullptr;
 
-    // if not enabled or not fence set-up always return true
+    // if fences are enabled but none selected fail pre-arm check
+    if ( enabled() && !present() ) {
+        fail_msg = "Fences enabled, but none selected";
+        return false;
+    }
+
+   // if not enabled or not fence set-up always return true
```
Comment (joshanne) [inline libraries/AC_Fence/AC_Fence.cpp:291]: Nitpick: not indented far enough.
```suggestion
    // if not enabled or not fence set-up always return true
```

**Review by joshanne — APPROVED** [review]
LGTM, excepting a nitpick.

File: libraries/AC_Fence/AC_Fence.cpp (hunk 2 above)
Comment (Hwurzburg) [reply libraries/AC_Fence/AC_Fence.cpp]: done

Comment (joshanne) [reply libraries/AC_Fence/AC_Fence.cpp]: 🙂

**Review by joshanne — APPROVED** [review]
LGTM :-)

**Review by tridge — APPROVED** [review]

File: libraries/AC_Fence/AC_Fence.cpp [hunk 3]
```diff
@@ -282,6 +282,12 @@ bool AC_Fence::pre_arm_check(const char* &fail_msg) const
 {
     fail_msg = nullptr;
 
+    // if fences are enabled but none selected fail pre-arm check
+    if ( enabled() && !present() ) {
```
Comment (rmackay9) [inline libraries/AC_Fence/AC_Fence.cpp:286]: can you remove the little space between "(" and "enabled()"?

Comment (Hwurzburg) [reply libraries/AC_Fence/AC_Fence.cpp]: done

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**